### PR TITLE
Metadata Manager : List Metadata API enhancement as per latest design docs

### PR DESCRIPTION
**Pull request by josephb160515cs**
**What type of PR is this?**
> /kind enhancement

**What this PR does / why we need it**:
CurrenListMetadataAPI doesnt support almost all the request parameters mentioned in the design docs.
This PR fetches the request parameters from the request from the api handler and sends to metadata manager m8s using gRPC
**Which issue(s) this PR fixes**:

Fixes #

**Test Report Added?**:
> /kind TESTED
* Limit & Offset is valid 
image

* Limit/Offset is invalid
image

**Special notes for your reviewer**:

### CODE REVIEW

File: api/pkg/metadata/service.go [hunk 1]
```diff
@@ -68,3 +74,47 @@ func (s *APIService) ListMetadata(request *restful.Request, response *restful.Re
...
+	offsetStr := request.PathParameter("offset")
+	bucketName := request.PathParameter("bucketName")
+	objectName := request.PathParameter("objectName")
+	sizeOfObject := request.PathParameter("sizeOfObject")
+	sizeOfBucket := request.PathParameter("sizeOfBucket")
+
+	offsetStr, limitStr = assignDefaultValues(offsetStr, limitStr)
+
+	// convert string offset to int32
+	offset, err := strconv.ParseInt(offsetStr, 10, 32)
```
Comment (PravinRanjan10) [inline api/pkg/metadata/service.go:93]: You may use  GetPaginationParam() function from common.go

File: api/pkg/metadata/service.go [hunk 2]
```diff
@@ -68,3 +74,47 @@ func (s *APIService) ListMetadata(request *restful.Request, response *restful.Re
...
+	if err != nil {
+		log.Errorf("Failed to parse offset as int32: \n", err)
+		return mt.ListMetadataRequest{}, err
+	}
+
+	// convert string limit to int32
+	limit, err := strconv.ParseInt(limitStr, 10, 32)
```
Comment (PravinRanjan10) [inline api/pkg/metadata/service.go:100]: same as above comment

File: metadata/proto/metadata.proto [hunk 3]
```diff
@@ -18,7 +18,23 @@ message SyncMetadataRequest {
...
+    string backendName = 4;
+    string region = 5;
+    // name of the bucket
+    string bucketName = 6;
+    // name of the object
+    string objectName = 7;
+    // the size of the object in <operator integer  unit> where operator is eq, gte,gt,lte,lt and unit is KB,MB,GB and integer is always not less than zero
+    string sizeOfObject = 8;
+    // the size of the bucket in <operator integer  unit> where operator is eq, gte,gt,lte,lt and unit is KB,MB,GB and integer is always not less than zero
+    string sizeOfBucket = 9;
```
Comment (PravinRanjan10) [inline metadata/proto/metadata.proto:37]: why string?, better to add as Integer.

File: metadata/proto/metadata.proto [hunk 4]
```diff
@@ -18,7 +18,23 @@ message SyncMetadataRequest {
...
+    string type = 3;
+    //name of the backend
+    string backendName = 4;
+    string region = 5;
+    // name of the bucket
+    string bucketName = 6;
+    // name of the object
+    string objectName = 7;
+    // the size of the object in <operator integer  unit> where operator is eq, gte,gt,lte,lt and unit is KB,MB,GB and integer is always not less than zero
+    string sizeOfObject = 8;
```
Comment (PravinRanjan10) [inline metadata/proto/metadata.proto:35]: why string?, better to add as Integer.

**Review by PravinRanjan10 — CHANGES_REQUESTED** [review]

File: api/pkg/metadata/service.go (hunk 1 above)
Comment (josephb160515cs) [reply api/pkg/metadata/service.go]: done

File: metadata/proto/metadata.proto (hunk 3 above)
Comment (josephb160515cs) [reply metadata/proto/metadata.proto:37]: done

File: api/pkg/metadata/service.go (hunk 2 above)
Comment (josephb160515cs) [reply api/pkg/metadata/service.go]: done

File: metadata/proto/metadata.proto (hunk 3 above)
Comment (josephb160515cs) [reply metadata/proto/metadata.proto]: As per discussion converted to integer and added another field for operation

File: metadata/proto/metadata.proto (hunk 4 above)
Comment (josephb160515cs) [reply metadata/proto/metadata.proto]: As per discussion converted to integer and added another field for operation

File: api/pkg/metadata/service.go [hunk 5]
```diff
@@ -51,20 +51,55 @@ func (s *APIService) SyncMetadata(request *restful.Request, response *restful.Re
...
+//* It returns ListMetadataRequest for ListMetaData API call
+func GetListMetaDataRequest(request *restful.Request) (listMetadataRequest mt.ListMetadataRequest, err error) {
+	typeOfCloudVendor := request.PathParameter("type")
+	backendName := request.PathParameter("backendName")
+	bucketName := request.PathParameter("bucketName")
+	objectName := request.PathParameter("objectName")
+	sizeOfObjectInBytes, err := common.GetSizeRequestParamAsInt64(request, "sizeOfObject")
+
+	if err != nil {
+		return mt.ListMetadataRequest{}, err
```
Comment (PravinRanjan10) [inline api/pkg/metadata/service.go:89]: Please avoid returing err messge directly to user. You may return.

type of sizeOfObject is invalid. It should be Integer type.

Comment (josephb160515cs) [reply api/pkg/metadata/service.go:89]: Done

**Review by PravinRanjan10 — APPROVED** [review]
LGTM

**Pull request merged**